src/main/java/org/ngafid/flights/process/Pipeline.java

```diff
@@ -297,7 +297,15 @@ public synchronized void addDerivedFile(String filename, byte[] data) throws IOE
         }
 
         Path zipFileSystemPath = derivedFileSystem.getPath(filename);
-        Files.createDirectories(zipFileSystemPath.getParent());
+        Path parentPath = zipFileSystemPath.getParent();
+        if (parentPath != null && !Files.exists(parentPath)) {
+            Files.createDirectories(parentPath);
+        }else{
+            String path = upload.getDerivedDirectory();
+            Path derivedPath = Paths.get(path);
+            Files.createDirectories(derivedPath);
+        }
+
```

Comment:
I don't know what the purpose of this is.

Reply:
It simply checks if the directory exists, and if it doesn't, it creates it. I removed else, since it is redundant.